commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt

```diff
@@ -46,6 +47,8 @@ var Submission.title: String?
 
 fun Submission.allFiles() = section.allFiles() + section.allSections().flatMap { it.allFiles() }
 fun Submission.getSectionByType(name: String) = section.allSections().first { it.type == name }
-fun Submission.libFileSections() = (section.allSections() + section).filterNot { it.libraryFile == null }
-fun Submission.allReferencedFiles() =
-    section.allReferencedFiles() + section.allSections().flatMap { it.allReferencedFiles() }
+fun Submission.libFileSections() = (section.allSections() + section).filterNot { it.libraryFile.isNullOrBlank() }
+
+fun ExtendedSubmission.allExtendedSections() = extendedSection.allExtendedSections() + extendedSection
```

Comment:
Same here I think we can separate extension by base type

Reply:
Done